python/src/spark_rapids_ml/classification.py

```diff
@@ -507,50 +537,89 @@ def _transformEvaluate(
+                ]
+            )
+            eval_metric_info = EvalMetricInfo(
+                eval_metric=transform_evaluate_metric.accuracy_like
+            )
+        # TBD: use toPandas and pandas df operations below
+        rows = (
+            super()
+            ._transform_evaluate_internal(dataset, schema, eval_metric_info)
+            .collect()
+        )
 
         num_models = (
             len(self._treelite_model) if isinstance(self._treelite_model, list) else 1
         )
 
-        tp_by_class: List[Dict[float, float]] = [{} for _ in range(num_models)]
-        fp_by_class: List[Dict[float, float]] = [{} for _ in range(num_models)]
-        label_count_by_class: List[Dict[float, float]] = [{} for _ in range(num_models)]
-        label_count = [0 for _ in range(num_models)]
+        if eval_metric_info.eval_metric == transform_evaluate_metric.accuracy_like:
+            tp_by_class: List[Dict[float, float]] = [{} for _ in range(num_models)]
+            fp_by_class: List[Dict[float, float]] = [{} for _ in range(num_models)]
+            label_count_by_class: List[Dict[float, float]] = [
+                {} for _ in range(num_models)
+            ]
+            label_count = [0 for _ in range(num_models)]
 
-        for i in range(num_models):
-            for j in range(self._num_classes):
-                tp_by_class[i][float(j)] = 0.0
-                label_count_by_class[i][float(j)] = 0.0
-                fp_by_class[i][float(j)] = 0.0
+            for i in range(num_models):
+                for j in range(self._num_classes):
+                    tp_by_class[i][float(j)] = 0.0
+                    label_count_by_class[i][float(j)] = 0.0
+                    fp_by_class[i][float(j)] = 0.0
 
-        for row in rows:
-            label_count[row.model_index] += row.total
-            label_count_by_class[row.model_index][row.label] += row.total
+            for row in rows:
+                label_count[row.model_index] += row.total
+                label_count_by_class[row.model_index][row.label] += row.total
+
+                if row.label == row.prediction:
+                    tp_by_class[row.model_index][row.label] += row.total
+                else:
+                    fp_by_class[row.model_index][row.prediction] += row.total
+
+            scores = []
+            for i in range(num_models):
+                metrics = MulticlassMetrics(
+                    tp=tp_by_class[i],
+                    fp=fp_by_class[i],
+                    label=label_count_by_class[i],
+                    label_count=label_count[i],
+                )
+                scores.append(metrics.evaluate(evaluator))
+        else:
+            # logLoss metric
+            label_count = [0 for _ in range(num_models)]
+            log_loss = [0.0 for _ in range(num_models)]
+            for row in rows:
+                label_count[row.model_index] += row.total
+                log_loss[row.model_index] += row.log_loss
```

Comment:
Apologies for my math. So we directly add up the partial log_loss, then get the final log_loss?

Reply:
that's correct.   And then divide by total number of examples to get the average log loss.